OpenRA.Mods.Common/Traits/Harvester.cs

```diff
@@ -414,7 +414,7 @@ public void ResolveOrder(Actor self, Order order)
 				self.QueueActivity(deliver);
 
 				foreach (var n in notify)
-					n.MovingToRefinery(self, order.TargetLocation, deliver);
+					n.MovingToRefinery(self, order.Target.Actor.Location, deliver);
```

Comment:
Can you please change this to:
```csharp
if (order.Type == TargetType.Actor)
	foreach (var n in notify)
		n.MovingToRefinery(self, order.Target.Actor, deliver);
```
and then adjust the `INotifyHarvesterAction` interface so that `MovingToRefinery` takes an `Actor` instead of a `CPos`?

In `AutoCarryable` you can then get the correct drop point to pass to `RequestTransport` by copying the `proc.Location + iao.DeliveryOffset` from `DeliverResources` (which is the only other place that requires non-trivial changes – just remove the `iao` bit and pass it the `proc` directly).